### PR TITLE
Showvariable paramter fix

### PR DESCRIPTION
**Pull request by KyleOndy**
Working on #403. I am unfamiliar with reflection so I am not sure I am even on the right path here. I will look into writing some tests shortly.

### CODE REVIEW

File: GitVersionExe/ArgumentParser.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public class ArgumentParser
     {
         static ArgumentParser()
         {
-            var fields = typeof(VariableProvider).GetFields(BindingFlags.Public | BindingFlags.Static);
+            var fields = typeof(VersionVariables).GetProperties();
             VersionParts = fields.Select(x => x.Name.ToLower()).ToArray();
```
Comment (JakeGinnivan) [inline GitVersionExe/ArgumentParser.cs]: Should actually be able to just do `new VariableProvider().AvailableVariables` now.

**Comment by KyleOndy** [comment]
I see I failed the CI build. I'll take a better look at this tonight.

**Comment by JakeGinnivan** [comment]
`Using directive is not required by the code and can be safely removed`

`using System.Reflection` is not needed anymore

File: GitVersionExe/ArgumentParser.cs (hunk 1 above)
Comment (KyleOndy) [inline GitVersionExe/ArgumentParser.cs]: I see an 'AvailableVariables' property contined in VersionVariables. To access that property I need to instantiate a new VersionVariables instance.

**Comment by JakeGinnivan** [comment]
One additional thing which would be good is to make a better error message when the variable doesn't exist.

At the moment:  
.\GitVersionExe\bin\Debug\GitVersion.exe /showvariable SemVer2  
`Failed to parse arguments: /showvariable SemVer2`

I would like

```
Variable SemVer2 doesn't exist, available variables:
....
list them all
....
```

Do you think you could do that as part of this PR? Otherwise this looks good

**Comment by JakeGinnivan** [comment]
@KyleOndy any thoughts on extending this PR to list available variables before I merge?

**Comment by KyleOndy** [comment]
Let me clean up what I have and get some tests written.

**Comment by JakeGinnivan** [comment]
@KyleOndy how did you go?

**Comment by JakeGinnivan** [comment]
Superseded by #482 

@KyleOndy Thanks for the contribution anyways :)

**Pull request closed without merging**